### PR TITLE
Remove Encoding#replicate

### PR DESCRIPTION
**Pull request by eregon**
https://bugs.ruby-lang.org/issues/18949

### CODE REVIEW

**Comment by eregon** [comment]
@soutaro is it OK to update to rbs 3.0.0.dev.1 here (for Ruby master)?
It's needed to get https://github.com/ruby/rbs/pull/1188

File: test/-ext-/string/test_too_many_dummy_encodings.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    assert_separately(%w[--disable=gems], "#{<<~"begin;"}\n#{<<~'end;'}")
+    begin;
+      require "-test-/string"
+      begin
+        1_000.times{|i| Bug::String.rb_define_dummy_encoding("R_#{i}") } # now 256 entries
+      rescue EncodingError => e
+        assert_match(/too many encoding/, e.message)
+      else
+        assert false
+      end
```
Comment (nobu) [inline test/-ext-/string/test_too_many_dummy_encodings.rb:16]:
```suggestion
      assert_raise_with_message(EncodingError, /too many encoding/) do
        1_000.times{|i| Bug::String.rb_define_dummy_encoding("R_#{i}") } # now 256 entries
      end
```

But I don't think this test is necessary.

Comment (eregon) [reply test/-ext-/string/test_too_many_dummy_encodings.rb]: Thanks. It's useful to check the error when this happens, and ko1 had such a test in https://github.com/ruby/ruby/pull/7079/files#diff-d573d0e3e36684ece0beeabd7dbf202b3d789d55f5992ebc9ec53d7c2970c366L164 and now we need to move it to use rb_define_dummy_encoding()

File: gems/bundled_gems [hunk 2]
```diff
@@ -11,6 +11,6 @@ net-pop         0.1.2   https://github.com/ruby/net-pop
 net-smtp        0.3.3   https://github.com/ruby/net-smtp
 matrix          0.4.2   https://github.com/ruby/matrix
 prime           0.1.2   https://github.com/ruby/prime
-rbs             2.8.3   https://github.com/ruby/rbs
+rbs             3.0.0.dev.1   https://github.com/ruby/rbs
```
Comment (nobu) [inline gems/bundled_gems:14]:
```suggestion
rbs             2.8.3  https://github.com/ruby/rbs 084fdb86387396eee08b2e525d648cae04d7b5e6
```

Comment (eregon) [reply gems/bundled_gems]: I picked https://github.com/ruby/rbs/pull/1188/commits/84526b58a88818661541f10de40074d63f298541 it should be enough

**Comment by eregon** [comment]
@mame typeprof seems to fail when using a newer `rbs`: https://github.com/ruby/ruby/actions/runs/3882979671/jobs/6623744675

**Pull request merged**